yaas/blueprint.py

```diff
@@ -1,4 +1,138 @@
 # coding: utf-8
 
+from __future__ import absolute_import
+from __future__ import print_function
+
+import argparse
+import inspect
+import json
+import pprint
+import requests
+import sys
+
+from . import config
+
+
 def command(parser, args):
-    pass
+    """ Interact with Ambari blueprints. """
+    subparsers = parser.add_subparsers(dest='subcommand')
+    subcommands = {
+        'add': _add,
+        'list': _list,
+        'remove': _remove,
+        'show': _show,
+        }
+    for name, fn in subcommands.items():
+        subparsers.add_parser(name, help=inspect.getdoc(fn))
+    subargs, extra = parser.parse_known_args(args)
+    subcommands[subargs.subcommand](
+        subparsers.choices[subargs.subcommand],
+        extra)
+
+
+def _add(parser, args):
+    """ Add a blueprint to the Ambari server. """
+    parser.add_argument(
+        '--validate-topology',
+        default='true',
+        help="Specify whether to validate a blueprint on upload.")
+    parser.add_argument(
+        'blueprint',
+        nargs='*',
+        help="Specify a blueprint to add to the Ambari server.")
+    subargs, extra = parser.parse_known_args(args)
+    blueprints = [] if len(subargs.blueprint) > 0 else [sys.stdin.read()]
+    for blueprint_file in subargs.blueprint:
+        with open(blueprint_file, 'r') as f:
+            blueprints.append(f.read())
+    for blueprint in blueprints:
+        blueprint = json.loads(blueprint)
+        response = requests.post(
+            config.href(
+                '/api/v1/blueprints/{name}'.format(
+                    name=blueprint['Blueprints']['blueprint_name'])),
+            data=json.dumps(blueprint),  # `json=blueprint` would be better, but it doesn't work!
```

Comment:
Python 2.7.10
Requests 2.5.0

This is the weird content-type thing I mentioned before.